bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -0,0 +1,339 @@
+            new Capabilities(3).primary().tiltOnClosed()                      .text("Vertical"),
+            new Capabilities(4).primary().tiltAnywhere().tilt180()            .text("Vertical Tilt 180°"),
+            new Capabilities(5)          .tiltAnywhere().tilt180()            .text("Tilt Only 180°"),
+            new Capabilities(6).primary()                                     .text("Top Down")                 .primaryStateInverted(),
+            new Capabilities(7).primary()                         .secondary().text("Top Down Bottom Up"),
+            new Capabilities(8).primary()                                     .text("Duolite Lift"),
+            new Capabilities(9).primary().tiltAnywhere()                      .text("Duolite Lift and Tilt 90°"),
+    // @formatter:on
+            new Capabilities()).stream().collect(Collectors.toMap(Capabilities::getValue, Function.identity()));
+
+    /*
+     * Database of known shade types and corresponding capabilities.
+     */
+    private static final Map<Integer, Type> TYPE_DATABASE = Arrays.asList(
+    // @formatter:off
+            new Type( 4).capabilities(0).text("Roman"),
+            new Type( 5).capabilities(0).text("Bottom Up"),
+            new Type( 6).capabilities(0).text("Duette"),
+            new Type( 7).capabilities(6).text("Top Down"),
+            new Type( 8).capabilities(7).text("Duette Top Down Bottom Up"),
+            new Type( 9).capabilities(7).text("Duette DuoLite Top Down Bottom Up"),
+            new Type(23).capabilities(1).text("Silhouette"),
+            new Type(42).capabilities(0).text("M25T Roller Blind"),
+            new Type(43).capabilities(1).text("Facette"),
+            new Type(44).capabilities(0).text("Twist"),
+            new Type(47).capabilities(7).text("Pleated Top Down Bottom Up"),
+            new Type(49).capabilities(0).text("AC Roller"),
+            new Type(51).capabilities(2).text("Venetian"),
+            new Type(54).capabilities(3).text("Vertical Slats Left Stack"),
+            new Type(55).capabilities(3).text("Vertical Slats Right Stack"),
+            new Type(56).capabilities(3).text("Vertical Slats Split Stack"),
+            new Type(62).capabilities(2).text("Venetian"),
+            new Type(69).capabilities(3).text("Curtain Left Stack"),
+            new Type(70).capabilities(3).text("Curtain Right Stack"),
+            new Type(71).capabilities(3).text("Curtain Split Stack"),
+            new Type(79).capabilities(8).text("Duolite Lift"),
+    // @formatter:on
+            new Type()).stream().collect(Collectors.toMap(Type::getValue, Function.identity()));
+
+    /**
+     * Base class that is extended by Type and Capabilities classes.
+     *
+     * @author Andrew Fiddian-Green - Initial Contribution
+     */
+    private static class Base {
+        protected int intValue = -1;
+        protected String text = "-- not in database --";
+
+        protected Integer getValue() {
```

Comment:
Is there a reason to not return int ?

Reply:
Not sure. I will check.

Reply:
Maybe for the map ?
I don't remember if a Map requires objects as key or if a simple basic type is ok too.

Reply:
That could be it. Maps do require objects rather than primitives.

Reply:
Ok, I close this comment.